TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/CapturedTrafficToHttpTransactionAccumulator.java

```diff
@@ -215,9 +257,14 @@ public void close() {
     }
 
     private void fireAccumulationsCallbacks(Accumulation accumulation) {
+        if (accumulation.trafficStreamsSorter.hasPending()) {
+            onTrafficStreamMissingOnExpiration.accept(accumulation);
+        }
         switch (accumulation.state) {
             case NOTHING_SENT:
-                handleEndOfRequest(accumulation);
+                if (!handleEndOfRequest(accumulation)) {
+                    return;
```

Comment:
What does this return achieve? Wouldn't we just complete this method after this switch case anyway?

Reply:
I'm not sure I follow this but its not a blocker for me either way